spring-integration-amqp/src/main/java/org/springframework/integration/amqp/inbound/AmqpInboundGateway.java

```diff
@@ -294,8 +294,9 @@ public void onMessage(final Message message, final Channel channel) throws Excep
 			catch (RuntimeException e) {
 				if (getErrorChannel() != null) {
 					setAttributesIfNecessary(message, null);
-					AmqpInboundGateway.this.messagingTemplate.send(getErrorChannel(), buildErrorMessage(null,
-							new ListenerExecutionFailedException("Message conversion failed", e, message)));
+					AmqpInboundGateway.this.messagingTemplate.send(getErrorChannel(), // NOSONAR not null
```

Comment:
Can't we extract `MessageChannel errorChannel = getErrorChannel();` variable instead?

To be honest I'm not a fan of this kind of `// NOSONAR` comments if we really can fix a smell.